### ISSUE TITLE
fatal errors in Stm23cubeIDE

### ISSUE THREAD

**Issue by shivashankarivr** [ShawnHymel/ei-keyword-spotting#4]
Hi, am getting the following errors when I try to build the code.

Description	Resource	Path	Location	Type
fatal error: edge-impulse-sdk/tensorflow/lite/c/builtin_op_data.h: No such file or directory	trained_model_compiled.cpp	/EdgeImpulse-Magikword-yes-no/ei-keyword-spotting/tflite-model	line 26	C/C++ Problem
fatal error: tensorflow/lite/micro/memory_planner/greedy_memory_planner.h: No such file or directory	greedy_memory_planner.cc	/EdgeImpulse-Magikword-yes-no/ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner	line 16	C/C++ Problem
fatal error: tensorflow/lite/micro/memory_planner/linear_memory_planner.h: No such file or directory	linear_memory_planner.cc	/EdgeImpulse-Magikword-yes-no/ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner	line 16	C/C++ Problem
make: *** [ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner/subdir.mk:22: ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner/greedy_memory_planner.o] Error 1	EdgeImpulse-Magikword-yes-no		 	C/C++ Problem
make: *** [ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner/subdir.mk:24: ei-keyword-spotting/edge-impulse-sdk/tensorflow/lite/micro/memory_planner/linear_memory_planner.o] Error 1	EdgeImpulse-Magikword-yes-no		 	C/C++ Problem
make: *** [ei-keyword-spotting/tflite-model/subdir.mk:19: ei-keyword-spotting/tflite-model/trained_model_compiled.o] Error 1	EdgeImpulse-Magikword-yes-no		 	C/C++ Problem
make: *** Waiting for unfinished jobs....	EdgeImpulse-Magikword-yes-no		 	C/C++ Problem

**Comment by ShawnHymel** [comment]
I just checked, and the nucleo-l476-keyword-spotting project builds fine on STM32CubeIDE 1.6.0 for me. Which version of STM32CubeIDE are you using? Which project are you attempting to build? Did you make any modifications to the project?

**Comment by shivashankarivr** [comment]
hi, am also using Stm32CubeIde 1.6.0. I am using the SensorTile box board(STEVAL-MKSBOX1V1).

**Comment by ShawnHymel** [comment]
I have not tried any of this with the SensorTile, so I don't know if it will work. Did you make any modifications to the code?

**Comment by shivashankarivr** [comment]
No, I didnt change anything in the code. I just copied all the C codes
whereever you have added into the code.

**Comment by ShawnHymel** [comment]
It looks like the build system can't find some of the libraries (such as the edge-impulse-sdk directory). If you're just copying and pasting code, that won't work. You need to import the whole project (or at the very least, import the required libraries, like edge-impulse-sdk). I'm not sure what "EdgeImpulse-Magikword-yes-no" is...that's not a directory in any of the example projects in this repo.

**Comment by ShawnHymel** [comment]
There are a lot of changes you'll need to make, as none of my projects were created for the SensorTile, and the microphone on the SensorTile PDM (whereas I used I2S in my examples). 

I do not have a complete tutorial on how to port the keyword spotting demos to different boards, but this section in the README should give you an idea on how to start: [https://github.com/ShawnHymel/ei-keyword-spotting#importing-the-edge-impulse-library-into-your-own-build-system](https://github.com/ShawnHymel/ei-keyword-spotting#importing-the-edge-impulse-library-into-your-own-build-system)

**Comment by shivashankarivr** [comment]
I really thanks a lot for your reply sir

I have done the below steps.
I created edge impulse c++ library also o created separately c++ code from
stm32cube IDE for sensor tile box.
I imported both edge impulse library and sesortile code into my workspace.

One thing i did was i copied all c code from urs amd copied in mine.

Now, I have 1 last doubt. If i change code in PDM,will it work sir?

**Comment by ShawnHymel** [comment]
I'm not sure if it will work--I have not tried to use a PDM microphone with this model nor have I tried running the model on a SensorTile. I recommend seeing if you can get inference to work with a static set of features (e.g. numerical array of raw audio values) before working on streaming audio from the PDM microphone.
